sds/src/secondary_validation/nir_checksum.rs

```diff
@@ -0,0 +1,75 @@
+use std::str::Chars;
+
+use crate::secondary_validation::Validator;
+
+pub struct NirChecksum;
+
+impl Validator for NirChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        let mut input_iter = regex_match.chars();
+
+        let digit: Option<i64> = get_next_digit_chars(&mut input_iter, 13)
+            .and_then(|chars| chars.into_iter().collect::<String>().parse().ok());
+        if digit.is_none() {
+            return false;
+        }
+
+        let checksum: Option<u8> = get_next_digit_chars(&mut input_iter, 2)
+            .and_then(|chars| chars.into_iter().collect::<String>().parse().ok());
+        if checksum.is_none() || get_next_digit_str(&mut input_iter).is_some() {
+            return false;
+        }
+
+        (97 - digit.unwrap() % 97) as u8 == checksum.unwrap()
```

Comment:
Where does `97` come from? If it's the ASCII value for `a`, it should probably be a constant like `const A_VALUE: u8 = 'a' as u8;`

Reply:
It's literaly the number 97

From [wikipedia page](https://fr.wikipedia.org/wiki/Num%C3%A9ro_de_s%C3%A9curit%C3%A9_sociale_en_France) (in French):

> clé de contrôle = complément à 97 du nombre formé par les 13 premiers chiffres du NIR modulo 97 
(complément au NIR pour la Sécurité sociale)

translation
> control key = complement to 97 of the number formed by the first 13 digits of the NIR modulo 97 
(complement to NIR for Social Security)